tests/components/vesync/test_humidifier.py

```diff
@@ -0,0 +1,217 @@
+"""Tests for the humidifer module."""
+
+from contextlib import nullcontext
+from unittest.mock import patch
+
+import pytest
+
+from homeassistant.components.humidifier import (
+    ATTR_HUMIDITY,
+    ATTR_MODE,
+    DOMAIN as HUMIDIFIER_DOMAIN,
+    SERVICE_SET_HUMIDITY,
+    SERVICE_SET_MODE,
+)
+from homeassistant.config_entries import ConfigEntryState
+from homeassistant.const import (
+    ATTR_ENTITY_ID,
+    SERVICE_TURN_OFF,
+    SERVICE_TURN_ON,
+    STATE_UNAVAILABLE,
+)
+from homeassistant.core import HomeAssistant
+from homeassistant.exceptions import HomeAssistantError, ServiceValidationError
+
+from tests.common import MockConfigEntry
+
+
+async def test_humidifier_state(
+    hass: HomeAssistant, humidifier_config_entry: MockConfigEntry
+) -> None:
+    """Test the resulting setup state is as expected for the platform."""
+
+    humidifier_id = "humidifier.humidifier_200s"
+    expected_entities = [
+        humidifier_id,
+        "sensor.humidifier_200s_humidity",
+    ]
+
+    assert humidifier_config_entry.state is ConfigEntryState.LOADED
+
+    for entity_id in expected_entities:
+        assert hass.states.get(entity_id).state != STATE_UNAVAILABLE
+
+    assert hass.states.get("sensor.humidifier_200s_humidity").state == "35"
+
+    state = hass.states.get(humidifier_id)
+
+    # ATTR_HUMIDITY represents the target_humidity which comes from configuration.auto_target_humidity node
+    assert state.attributes.get(ATTR_HUMIDITY) == 40
+
+
+@patch(
+    "homeassistant.components.vesync.humidifier.VeSyncHumidifierHA.schedule_update_ha_state"
+)
```

Comment:
Let's not patch this out, rather check the state machine if the state updated

Reply:
Based on my reading of https://developers.home-assistant.io/docs/core/entity/, I think it was incorrect to call schedule_update_ha_state since nothing has changed in the entity. One could force a refresh but I think it would be okay for the state to updated in the next scheduled update.

I have removed the schedule_update_ha_state call itself and adjusted the tests.